File: backend/internal/entities/item.go
package entities

import (
	"github.com/google/uuid"
)

type Item struct {
	ID            uuid.UUID `gorm:"type:uuid;primaryKey" json:"item_id"`
	Name          string    `json:"name"`
	Description   string    `json:"description"`
	ItemTypeID    uuid.UUID `gorm:"type:uuid" json:"item_type_id"`
	RarityID      uuid.UUID `gorm:"type:uuid" json:"rarity_id"`
	RequiredLevel int       `json:"required_level"`
	MaxStack      int       `json:"max_stack"`

	ItemType  ItemType      `gorm:"foreignKey:ItemTypeID;references:ID" json:"item_type"`   // this.ItemTypeID -> ItemType.ID
	Rarity    Rarity        `gorm:"foreignKey:RarityID;references:ID" json:"rarity"`        // this.RarityID -> Rarity.ID
	ItemStats ItemLevelStat `gorm:"foreignKey:ID;references:ItemID" json:"item_level_stat"` // this.ID -> ItemLevelStat.ItemID
}

// func (i *Item) BeforeCreate(tx *gorm.DB) (err error) {
// 	i.ID = uuid.New()
// 	return
// }

// Tested

File: frontend/src/types/item.ts
import { ItemLevelStat } from "./itemLevelStat";
import { ItemType } from "./itemType";
import { Rarity } from "./rarity";

export interface Item {
  item_id: string;
  name: string;
  description: string;
  item_type_id: string;
  rarity_id: string;
  required_level: number;
  max_stack: number;

  item_type: ItemType;
  rarity: Rarity;
  item_level_stat: ItemLevelStat;
}
